python/triton/language/semantic.py

```diff
@@ -795,96 +795,196 @@ def str_to_eviction_policy(eviction_policy):
     return eviction
 
 
+def str_to_padding_option(padding_option):
+    padding = None  # default
+    if padding_option:
+        if padding_option == "zero":
+            padding = ir.PADDING_OPTION.PAD_ZERO
+        elif padding_option == "nan":
+            padding = ir.PADDING_OPTION.PAD_NAN
+        else:
+            raise ValueError(f"Padding option {padding_option} not supported")
+    return padding
+
+
+def canonicalize_boundary_check(boundary_check, tile_shape):
+    if boundary_check:
+        if not hasattr(boundary_check, "__iter__"):
+            boundary_check = [boundary_check]
+        boundary_check = [elem.value if isinstance(elem, tl.constexpr) else elem for elem in boundary_check]
+        for dim in boundary_check:
+            assert isinstance(dim, int) and 0 <= dim < len(tile_shape)
+        assert len(boundary_check) > 0
+        assert len(boundary_check) == len(set(boundary_check)), "Duplicate dimension in `boundary_check`"
+        return sorted(boundary_check)
+    return tuple()
+
+
 def load(ptr: tl.tensor,
```

Comment:
I'm not convinced the load and store function need to be so complicated. You could have two functions _unstructured_load (exactly the old code path) and _structured_load (new code path). Peak the right code path depending on the type of pointer. tensor<ptr> -> _unstructured and ptr<tensor> -> structured

Reply:
Ditto.